partner_credit_limit/model/partner.py

```diff
@@ -58,18 +58,15 @@ def _get_overdue_credit(self):
             # credit = 0.0
             debit_maturity, credit_maturity = 0.0, 0.0
             for line in movelines:
-                if line.date_maturity and line.partner_id.grace_payment_days:
+                limit_day = line.date_maturity
```

Comment:
Maybe you could delete other `if` using other logic way and reduce the mccabe rate.
```python
limit_day = fields.Datetime.from_string(line.date_maturity) + timedelta(days=line.partner_id.grace_payment_days or 0)
```